adobe_vipm/flows/sync.py

```diff
@@ -66,178 +67,158 @@ def sync_agreement_prices(mpt_client, agreement, dry_run, adobe_client, customer
-
-        to_update = []
-
-        for subscription in subscriptions:
-            if subscription["status"] == "Terminated":
-                continue
-
-            subscription = get_agreement_subscription(mpt_client, subscription["id"])
-            adobe_subscription_id = subscription["externalIds"]["vendor"]
-
-            adobe_subscription = adobe_client.get_subscription(
-                authorization_id,
-                customer_id,
-                adobe_subscription_id,
-            )
+    authorization_id = agreement["authorization"]["id"]
+    customer_id = get_adobe_customer_id(agreement)
+    currency = agreement["listing"]["priceList"]["currency"]
+    product_id = agreement["product"]["id"]
+    subscriptions = agreement["subscriptions"]
 
-            if adobe_subscription["status"] == STATUS_SUBSCRIPTION_TERMINATED:
-                logger.info(
-                    f"Skipping subscription {subscription['id']}. It is terminated"
-                )
-                continue
+    commitment = get_3yc_commitment(customer)
+    commitment_start_date = None
+    if (
+        commitment
+        and commitment["status"] in (STATUS_3YC_COMMITTED, STATUS_3YC_ACTIVE)
+        and date.fromisoformat(commitment["endDate"]) >= date.today()
+    ):
+        commitment_start_date = date.fromisoformat(commitment["startDate"])
 
-            actual_sku = adobe_subscription["offerId"]
+    coterm_date = customer["cotermDate"]
 
-            to_update.append((subscription,
-                              adobe_subscription,
-                              get_sku_with_discount_level(actual_sku, customer)))
+    to_update = []
 
-        skus = [item[2] for item in to_update]
+    for subscription in subscriptions:
+        if subscription["status"] == "Terminated":
+            continue
 
-        prices = get_sku_price(customer, skus, product_id, currency)
+        subscription = get_agreement_subscription(mpt_client, subscription["id"])
+        adobe_subscription_id = subscription["externalIds"]["vendor"]
 
-        for subscription, adobe_subscription, actual_sku in to_update:
-            if actual_sku not in prices:
-                logger.error(
-                    f"Skipping subscription {subscription['id']} "
-                    f"because the sku {actual_sku} is not in the prices"
-                )
-                missing_prices_skus.append(actual_sku)
-                continue
+        adobe_subscription = adobe_client.get_subscription(
+            authorization_id,
+            customer_id,
+            adobe_subscription_id,
+        )
 
-            line_id = subscription["lines"][0]["id"]
-            lines = [
-                {
-                    "price": {"unitPP": prices[actual_sku]},
-                    "id": line_id,
-                }
-            ]
+        if adobe_subscription["status"] == STATUS_SUBSCRIPTION_TERMINATED:
+            logger.info(f"Skipping subscription {subscription['id']}. It is terminated")
```

Comment:
I know that's not your code, but worth to add it. Could you please add to the log message, that it is terminated on Adobe :-)